cloudinit/net/activators.py

```diff
@@ -272,15 +279,19 @@ def search_activator(
         raise ValueError(
             "Unknown activators provided in priority list: %s" % unknown
         )
-
-    return [activator for activator in priority if activator.available(target)]
+    activator_classes = [NAME_TO_ACTIVATOR[name] for name in priority]
+    return [
+        activator_cls
+        for activator_cls in activator_classes
+        if activator_cls.available(target)
+    ]
 
 
 def select_activator(priority=None, target=None) -> Type[NetworkActivator]:
     found = search_activator(priority, target)
```

Comment:
With the typing additions to the `search_activators()` parameters, Pyright warns here.